src/autopas/containers/LeavingParticleCollector.h

```diff
@@ -9,38 +9,76 @@
- * eventually merge both loops?
  */
 template <class ContainerType>
 std::vector<typename ContainerType::ParticleType> collectParticlesAndMarkNonOwnedAsDummy(ContainerType &container) {
-  // First mark halo particles as dummy!
-  for (auto iter = container.begin(autopas::IteratorBehavior::halo); iter.isValid(); ++iter) {
-    iter->setOwnershipState(OwnershipState::dummy);
-  }
+  using namespace autopas::utils::ArrayMath::literals;
+
+  const auto upperBoundForMisplacement = container.getCutoff();
+  const auto containerDimensions = container.getBoxMax() - container.getBoxMin();
+  const auto lowerHaloCorner = container.getBoxMin() - upperBoundForMisplacement;
+  const auto upperHaloCorner = container.getBoxMax() + upperBoundForMisplacement;
+  // We need to extend these boundaries into the container because we need to include cells that still hold particles,
+  // which now have left the cells region towards the outside of the container.
+  // travelled beyond that and skin can be 0.
+  const auto lowerInnerCorner = container.getBoxMin() + upperBoundForMisplacement;
+  const auto upperInnerCorner = container.getBoxMax() - upperBoundForMisplacement;
+
+  // volumes describing the halo regions on the six faces of the cuboid shaped container.
+  const std::array<std::tuple<decltype(lowerHaloCorner), decltype(upperHaloCorner)>, 6> haloVolumes{{
+      // left + lower edge
+      {{lowerHaloCorner[0], lowerInnerCorner[1], lowerHaloCorner[2]},
+       {lowerInnerCorner[0], upperInnerCorner[1], upperInnerCorner[2]}},
+      // right + upper edge
+      {{upperInnerCorner[0], lowerInnerCorner[1], lowerInnerCorner[2]},
+       {upperHaloCorner[0], upperInnerCorner[1], upperHaloCorner[2]}},
+      // top + left edge
+      {{lowerHaloCorner[0], lowerInnerCorner[1], upperInnerCorner[2]},
+       {upperInnerCorner[0], upperInnerCorner[1], upperHaloCorner[2]}},
+      // bottom + right edge
+      {{lowerInnerCorner[0], lowerInnerCorner[1], lowerHaloCorner[2]},
+       {upperHaloCorner[0], upperInnerCorner[1], lowerInnerCorner[2]}},
+      // front with all edges and corners
+      {lowerHaloCorner, {upperHaloCorner[0], lowerInnerCorner[1], upperHaloCorner[2]}},
+      // back with all edges and corners
+      {{lowerHaloCorner[0], upperInnerCorner[1], lowerHaloCorner[2]}, upperHaloCorner},
+  }};
 
-  std::vector<typename ContainerType::ParticleType> leavingParticles;
-  // Collect leaving particles
-  for (auto iter = container.begin(autopas::IteratorBehavior::owned); iter.isValid(); ++iter) {
-    if (not utils::inBox(iter->getR(), container.getBoxMin(), container.getBoxMax())) {
-      leavingParticles.push_back(*iter);
-      iter->setOwnershipState(OwnershipState::dummy);
+  std::vector<typename ContainerType::ParticleType> leavingParticles{};
+#ifdef AUTOPAS_OPENMP
+#pragma omp declare reduction(vecMergeParticle : std::vector<typename ContainerType::ParticleType> : omp_out.insert(omp_out.end(), omp_in.begin(), omp_in.end()))
```

Comment:
Add a brief one-line explaination of `vecMergeParticles`

Reply:
done d04ac70d52e340f77c38cf5429b5a76f3e3243e8